inference-engine/thirdparty/clDNN/src/primitive_inst.cpp

```diff
@@ -161,6 +182,32 @@ primitive_inst::primitive_inst(network& network, program_node const& node, bool
     }
 }
 
+void primitive_inst::allocate_internal_buffers(void) {
+    if (_impl == nullptr) return;
+    const auto& ibuf_info = _impl->get_internal_buffer_info();
+    if (ibuf_info.sizes.empty()) return;
+
+    auto& engine = get_network().get_engine();
+    auto alloc_type = allocation_type::usm_host;
+
+    if (std::accumulate(ibuf_info.sizes.begin(), ibuf_info.sizes.end(), 0) <= engine.get_device_info().max_alloc_mem_size * 0.85) {
```

Comment:
this 0.85 constant looks like a hack. If it's really needed to avoid some issues, please add a comment describing why we need it, and why it's equal to 0.85.

Reply:
I agree. I'd like to find a rationale criteria of memory management, but I couldnt found.

Reply:
Does the issue really happen if **sum** of all intermediate buffers exceed some value or for single allocation bigger than something? I'm just trying to understand why it doesn't fail for all tensors then if it depends on total allocation size.

Reply:
In the target network, actually only one internal buffer is extremely big and the other is very small. I could not reproduce same failure with distributed amount of the memory to several internal buffer where the total is > 0.85% max. BTW the failure happens only when we run multi streams. For one stream execution, it is okay to run with the internal buffer > 0.85% of max. I am suspecting somehting is happening b/w streams...

Reply:
What about total memory consumption with single stream vs multiple streams? Can the issue be connected with exceeding available global memory size for multi-stream scenario?

Reply:
Global memory size : 3.144 GB 
Preferred total size of global vars : 1.57 GB 

(while for one stream)
[passing]
internal buf : 1.48 GB 
input tensors: 0.62 MB 
[fails]
internal buf : 1.44 GB
input tensors: 0.6 GB

Reply:
I will fix above description. even with one stream execution, 1.48GB case is failing and 1.44 GB case is passing. Let me do more investigation.

Reply:
I fixed memory recording logic to record host and device mem separately, but the device mem usage was not that different.
- with original logic :  7.38 GB right after internal buffers are allocated
- with fixed logic : 7.34 GB right after internal buffers are allocated (this is correct amount) 

The reason why the gap is not that large is becuase the majority of the host allocations are Freed before that. 
So, 7.34 GB is being used. 
_GPU_Debug: Allocate 1489488000 bytes of usm_device allocation type (current=7349808816; max=7349808816)_

Reply:
Also it is really strange.. : (
When I tried before, batch31 was passing. However now it is not passing too. I ran with same code with before, but the result is different.  Wondering whether the passing situation can be changed by time to tome w.r.t the HW status? ... hmm

Reply:
More interesting thing. b32 was passing once, after I cleared the dmesg logs. I dont think this is direct relevant, but the thing is that the trend depends on execution.

Reply:
The conclusion from alignment with driver's condition:
- The abort was coming from the old driver version 
- Driver's requirement : total device memory allocation for a kernel (including input, internal buffer, output) should be less than CL_DEVICE_GLOBAL_MEM_SIZE

Reply:
updated the policy w.r.t the driver's requirement